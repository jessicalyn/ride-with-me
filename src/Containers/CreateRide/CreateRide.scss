.create-ride-component {
  box-shadow: $box-shadow;
  border: 2px solid black;
  width: 80%;
  margin: 20px auto;
  background-color: white
}