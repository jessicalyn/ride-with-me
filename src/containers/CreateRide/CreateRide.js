import React, { Component } from 'react';

export class CreateRide extends Component {

  render() {
    return(
      <div className="create-ride-container">
        <h3>Create a Ride</h3>
        <form>
          <div>
            <label>Start City</label>
            <input type="text"></input>
          </div>
          <div>
            <label>End City</label>
            <input type="text"></input>
          </div>
          <div>
            <label>Start Date</label>
            <input type="text"></input>
          </div>
          <div>
            <label>Compensation</label>
            <input type="text"></input>
          </div>
          <div>
            <label>Available Seats</label>
            <input type="text"></input>
          </div>
          <div>
            <label>Short Description</label>
            <input type="text"></input>
          </div>
          <button>Add Ride</button>
        </form>
      </div>
    )
  }
}
