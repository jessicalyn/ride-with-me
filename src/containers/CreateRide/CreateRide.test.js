import React from 'react';
import { shallow } from 'enzyme';
import { CreateRide } from './CreateRide';

describe('CreateRide', () => {

  describe('CreateRide container', () => {
    let wrapper
    const mockDescription = "Going to visit Grandma"
    const mockUUID = "1320498-2349-3049"
    const mockStartCityId = 5
    const mockEndCityId = 4
    const mockPrice = 200
    const mockTotalSeats = 5
    const mockDepartureDate = "2019-05-14"
    const mockUser = { id: 2, uuid: "1320498-2349-3032432449" }
    const mockCities = [{ id: 1, name: "Denver"}, { id: 2, name: "San Diego"}]
    const mockState = {
      driverUuid: "1320498-2349-3032432449",
      startCityId: 1,
      endCityId: 2,
      description: "Going to Grandmas",
      price: "$300",
      totalSeats: 3,
      departureDate: "2019-05-14" 
    }
    
    beforeEach(() => {
      wrapper = shallow(
        <CreateRide
          description={mockDescription}
          driverUuid={mockUUID}
          startCityId={mockStartCityId}
          endCityId={mockEndCityId}
          price={mockPrice}
          totalSeats={mockTotalSeats}
          departureDate={mockDepartureDate}
          user={mockUser}
          cities={mockCities}
        />)
    })
    
    it('should match the snapshot with all data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    })
  
    it.skip('should have proper default state', () => {
      expect(wrapper.state()).toEqual(mockState)
    })
  })
})