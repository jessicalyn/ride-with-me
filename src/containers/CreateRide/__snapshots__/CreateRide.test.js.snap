// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRide CreateRide container should match the snapshot with all data passed in 1`] = `undefined`;
